File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int r_score = count(s.begin(), s.end(), '1');
        int l_score = 0;
        int max_score = 0;
        int score = 0;
        int n = s.size();
        for(int i=0; i<n-1; i++){
            if(s[i] == '0'){
                l_score ++;
            }
            if(s[i] == '1'){
                r_score --;
            }
            score = l_score + r_score;
            max_score = max(score, max_score);
        }
        return max_score;
        
    }
};

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        r_score = s.count("1")
        l_score = 0
        score = r_score + l_score
        max_score = 0
        for i in range(len(s) -1):
            if s[i] == "0":
                l_score += 1
            if s[i] == "1":
                r_score -= 1
            score = l_score + r_score
            max_score = max(score, max_score) 
        return max_score          


        